homeassistant/components/todoist/calendar.py

```diff
@@ -475,16 +475,16 @@ def create_todoist_task(self, data: Task):
             end = dt.parse_datetime(
                 data.due.datetime if data.due.datetime else data.due.date
             )
```

Comment:
The start date is set to the user's local time here:
https://github.com/home-assistant/core/blob/6ae99cd2750c9769327616d646cb44051af9bfb7/homeassistant/components/todoist/calendar.py#L454

So we need to make sure this end date is also in the user's local time so that we aren't comparing local time against utc.

Reply:
FWIW i removed this restriction in a follow up PR as there are valid use cases for these differences in calendar applications. That said, i still think this is worth changing to local time.